analyzers/integration_tests/project_initializer/node/node_initializer.go

```diff
@@ -0,0 +1,153 @@
+package node
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fossas/fossa-cli/analyzers/integration_tests/testtools"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/files"
+)
+
+type Project struct {
+	name   string
+	url    string
+	commit string
+	env    map[string]string
+}
+
+type NodeInitializer struct {
+	projects []Project
+}
+
+func New() NodeInitializer {
+	return NodeInitializer{
+		projects: projects,
+	}
+}
+
+func (n NodeInitializer) BuildAll() error {
+	testDir := n.FixtureDirectory()
+	testDirExists, err := files.ExistsFolder(testDir)
+	if err != nil {
+		return err
+	}
+	if testDirExists {
+		println(testDir + "already exists, skipping initialization step")
+		return nil
+	}
+
+	err = os.Mkdir(testDir, 0700)
+	if err != nil {
+		return err
+	}
+
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(n.projects))
+
+	for _, project := range n.projects {
```

Comment:
I think _this_ is the actual logic that will be common between packages. It would be more useful to provide `func CloneProjects(dir string, projects []Project) error` as a testing tool.